multiboot2/src/lib.rs

```diff
@@ -308,8 +313,41 @@ impl BootInformation {
         unsafe { &*self.inner }
     }
 
-    fn get_tag(&self, typ: TagType) -> Option<&Tag> {
-        self.tags().find(|tag| tag.typ == typ)
+    /// Public getter to find any tag type you are looking for.
+    ///
+    /// # Specified or Custom Tags
+    /// The Multiboot2 specification specifies a list of tags, see [`TagType`]. However, it doesn't
+    /// forbid to use custom tags. Because of this, there exists the [`SpecifiedOrCustomTagType`]
+    /// abstraction. **It is recommended** to use this getter only for custom tags. For specified
+    /// tags, use getters, such as [`Self::efi_64_ih`].
+    ///
+    /// ## Use Custom Tags
+    /// The following example shows how you may use this interface to parse custom tags from
+    /// the MBI.
+    ///
+    /// ```ignore
+    /// use multiboot2::SpecifiedOrCustomTagTypeSerialized;
+    /// #[repr(C, align(8))]
+    ///     struct CustomTag {
+    ///     // new type from the lib: has repr(u32)
+    ///     tag: SpecifiedOrCustomTagTypeSerialized,
+    ///     size: u32,
+    ///     // begin of inline string
+    ///     name: u8,
+    /// }
+    ///
+    /// let tag = bi
+    ///     // this function is now public!
+    ///     .get_tag(0x1337.into())
+    ///     .unwrap()
+    ///     // type definition from end user. must be Sized!
+    ///     .cast_tag::<CustomTag>();
+    /// let name = &tag.name as *const u8 as *const c_char;
+    /// let str = unsafe { CStr::from_ptr(name).to_str().unwrap() };
+    /// assert_eq!(str, "name");
+    /// ```
+    pub fn get_tag(&self, typ: SpecifiedOrCustomTagType) -> Option<&Tag> {
```

Comment:
I think this could take a `impl Into<SpecifiedOrCustomTagType` and then do the `into` in here to avoid all the `into` spam?

Reply:
oh, that's a good idea! thanks!